proxy_wasm_api.h

```diff
@@ -746,6 +746,14 @@ inline WasmResult getBufferStatus(WasmBufferType type, size_t *size, uint32_t *f
   return proxy_get_buffer_status(type, size, flags);
 }
 
+inline WasmResult setBuffer(WasmBufferType type, size_t start, size_t length, StringView data,
+                            size_t *new_size = nullptr) {
+  auto result = proxy_set_buffer_bytes(type, start, length, data.data(), data.size());
+  if (result == WasmResult::Ok && new_size)
+    *new_size += -length + data.size();
```

Comment:
`length` or `-length`?
Will the - in front of the `length` generate a very large number?

Reply:
I should add a test.  The amount in length is removed from the initial size.  If length is > the initial size then result != WasmResult::Ok.  So, this should be safe AFAICT.